### PR TITLE
Correct use of variable-speed DX cooling coil WB inlet temp during dry-coil operation.

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
To model the DX cooling coil, the inlet wet-bulb temperature is used as an input to the capacity as a function of temperature and EIR as a function of temperature performance curves. When a dry-coil condition exists, the model increases inlet air wet-bulb temperature until the solution converges. This new wet-bulb temperature is then used in the CAPfT and EIRfT performance curves. The variable-speed DX cooling coil did not use this new web-bulb temperature. This change corrects #6459.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by rraustad** [comment]
PR #6449 should be merged before this PR.

**Comment by rraustad** [comment]
Need to merge develop here when PR #6449 gets merged. Look for conflicts and correct as needed.

**Comment by rraustad** [comment]
I merged dev and am testing for irregularities.

**Comment by Myoldmopar** [comment]
Thanks @rraustad!!! Then go with the 🛥 and ☂️ and 🍸 !!

File: src/EnergyPlus/VariableSpeedCoils.cc [hunk 1]
```diff
@@ -1299,7 +1299,7 @@ namespace VariableSpeedCoils {
 			SetupOutputVariable( "Cooling Coil Total Cooling Energy", OutputProcessor::Unit::J, VarSpeedCoil( DXCoilNum ).EnergyLoadTotal, "System", "Summed", VarSpeedCoil( DXCoilNum ).Name, _, "ENERGYTRANSFER", "COOLINGCOILS", _, "System" );
 			SetupOutputVariable( "Cooling Coil Sensible Cooling Energy", OutputProcessor::Unit::J, VarSpeedCoil( DXCoilNum ).EnergySensible, "System", "Summed", VarSpeedCoil( DXCoilNum ).Name );
 			SetupOutputVariable( "Cooling Coil Latent Cooling Energy", OutputProcessor::Unit::J, VarSpeedCoil( DXCoilNum ).EnergyLatent, "System", "Summed", VarSpeedCoil( DXCoilNum ).Name );
-			SetupOutputVariable( "Cooling Coil Source Side Heat Transfer Energy", OutputProcessor::Unit::J, VarSpeedCoil( DXCoilNum ).EnergySource, "System", "Summed", VarSpeedCoil( DXCoilNum ).Name, _, "PLANTLOOPCOOLINGDEMAND", "COOLINGCOILS", _, "System" );
+			SetupOutputVariable( "Cooling Coil Source Side Heat Transfer Energy", OutputProcessor::Unit::J, VarSpeedCoil( DXCoilNum ).EnergySource, "System", "Summed", VarSpeedCoil( DXCoilNum ).Name );
```
Comment (rraustad) [inline src/EnergyPlus/VariableSpeedCoils.cc:1302]: This is air loop equipment, should not report to plant energy

File: src/EnergyPlus/VariableSpeedCoils.cc [hunk 2]
```diff
@@ -3932,15 +3932,16 @@ namespace VariableSpeedCoils {
 			RatedInletEnth = PsyHFnTdbW( RatedInletAirTemp, RatedInletAirHumRat );
 			CBFRated = AdjustCBF( VarSpeedCoil( DXCoilNum ).MSRatedCBF( NormSpeed ), VarSpeedCoil( DXCoilNum ).MSRatedAirMassFlowRate( NormSpeed ), RatedAirMassFlowRate );
 			if ( CBFRated > 0.999 ) CBFRated = 0.999;
-			AirMassFlowRatio = VarSpeedCoil( DXCoilNum ).RatedAirVolFlowRate / VarSpeedCoil( DXCoilNum ).MSRatedAirVolFlowRate( NormSpeed );
+			AirMassFlowRatio = RatedAirMassFlowRate / VarSpeedCoil( DXCoilNum ).MSRatedAirMassFlowRate( NormSpeed );
```
Comment (rraustad) [inline src/EnergyPlus/VariableSpeedCoils.cc:3935]: mass ratio based on mass not vol, really the same thing

File: src/EnergyPlus/VariableSpeedCoils.cc [hunk 3]
```diff
@@ -4410,6 +4411,12 @@ namespace VariableSpeedCoils {
 					WaterMassFlowRatio = SourceSideMassFlowRate / VarSpeedCoil( DXCoilNum ).DesignWaterMassFlowRate;
 				}
 
+				CBFSpeed = AdjustCBF( VarSpeedCoil( DXCoilNum ).MSRatedCBF( SpeedCal ), VarSpeedCoil( DXCoilNum ).MSRatedAirMassFlowRate( SpeedCal ), LoadSideMassFlowRate );
+
+				if ( CBFSpeed > 0.999 ) CBFSpeed = 0.999;
+
+				CalcTotCapSHR_VSWSHP( LoadSideInletDBTemp, LoadSideInletHumRat, LoadSideInletEnth, LoadSideInletWBTemp, AirMassFlowRatio, WaterMassFlowRatio, LoadSideMassFlowRate, CBFSpeed, VarSpeedCoil( DXCoilNum ).MSRatedTotCap( SpeedCal ), VarSpeedCoil( DXCoilNum ).MSCCapFTemp( SpeedCal ), VarSpeedCoil( DXCoilNum ).MSCCapAirFFlow( SpeedCal ), VarSpeedCoil( DXCoilNum ).MSCCapWaterFFlow( SpeedCal ), 0.0, 0, 0, 0, QLoadTotal1, QLoadTotal2, QLoadTotal, SHR, SourceSideInletTemp, VarSpeedCoil( DXCoilNum ).InletAirPressure, 0.0, 1, VarSpeedCoil( DXCoilNum ).capModFacTotal );
+
```
Comment (rraustad) [inline src/EnergyPlus/VariableSpeedCoils.cc:4419]: Just rearranged this to give the result required

File: src/EnergyPlus/VariableSpeedCoils.cc [hunk 4]
```diff
@@ -6458,7 +6465,7 @@ namespace VariableSpeedCoils {
 		Real64 const InletDryBulb, // inlet air dry bulb temperature [C]
 		Real64 const InletHumRat, // inlet air humidity ratio [kg water / kg dry air]
 		Real64 const InletEnthalpy, // inlet air specific enthalpy [J/kg]
-		Real64 const InletWetBulb, // inlet air wet bulb temperature [C]
+		Real64 & InletWetBulb, // inlet air wet bulb temperature [C]
```
Comment (rraustad) [inline src/EnergyPlus/VariableSpeedCoils.cc:6468]: no longer const, can be modified in function call

File: tst/EnergyPlus/unit/HVACUnitarySystem.unit.cc [hunk 5]
```diff
@@ -2226,7 +2226,7 @@ TEST_F( EnergyPlusFixture, SetOnOffMassFlowRateTest )
 	UnitarySystem( UnitarySysNum ).Humidistat = false;
 	UnitarySystem( UnitarySysNum ).DehumidControlType_Num = DataSizing::None;
 
-	// cycling fan mode should drop to 0 flow rate only below speed = 1
```
Comment (rraustad) [inline tst/EnergyPlus/unit/HVACUnitarySystem.unit.cc:2229]: and change unit tests to conform to changes here

**Comment by Myoldmopar** [comment]
Clangs already running over here, if Marvin is happy I'll be happy to merge.  Thanks for the last minute effort here.

**Comment by rraustad** [comment]
Unit tests in debug mode VS community 2015 passed. Geeeze does FluidPropertiesRefData take a long time, I might be back from vacation before this finishes.

**Comment by rraustad** [comment]
It's the VS coil example files that have diffs. This is not a surprise.

**Pull request merged**